components/core/src/clp/StreamingReader.hpp

```diff
@@ -0,0 +1,411 @@
+#ifndef CLP_S3_STREAMING_READER_HPP
+#define CLP_S3_STREAMING_READER_HPP
```

Comment:
According to the naming convention, can we update it to `CLP_STREAMING_READER_HPP` or create a directory `src/clp/s3`?